src/modules/storage/LedgerStorage.ts

```diff
@@ -164,11 +164,11 @@ export class LedgerStorage extends Storages
         return new Promise<void>((resolve, reject) =>
         {
             (async () => {
+                let block: Block = new Block();
+                block.parseJSON(data);
+                await this.begin();
```

Comment:
`block.parseJSON(data);` and `await this.begin()` 
Exception may occur here as well.
`reject()` should be called when an exception occurs here.

Reply:
Yes, I think so, too.
I think `let block: Block = new Block(); block.parseJSON(data);` can be move as `try`.

Reply:
I have confirmed that when an exception occurs in the code below, it is not handled normally.
``` TypeScript
        function testThrowError (value: number): Promise<number>
        {
            return new Promise<number>((resolve, reject) => {
                if (value === 0)
                    throw new Error("error");

                resolve(value*value);
            })
        }

        try
        {
            (async () =>
            {
                await testThrowError(0);
            })();
        }
        catch (e)
        {
            console.log(e);
        }
```

When I changed it as follows, I checked that it was handled normally.
```
        (async () =>
        {
            try
            {
                await testThrowError(0);
            }
            catch (e)
            {
                console.log(e);
            }
        })();
```